tests/integration/pilot/namespace_isolation_test.go

```diff
@@ -0,0 +1,66 @@
+package pilot
+
+import (
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/scopes"
+	"testing"
+	"time"
+)
+
+const isolationSidecar = `
+apiVersion: networking.istio.io/v1alpha3
+kind: Sidecar
+metadata:
+  name: isolate-ns
+spec:
+  egress:
+  - hosts:
+    - "./*"
+    - "istio-system/*"
+`
+
+func TestPushTriggerIsolation(t *testing.T) {
+	framework.NewTest(t).
+		Features("traffic.sidecar.isolation").
+		Run(func(ctx framework.TestContext) {
+			ns1 := namespace.NewOrFail(ctx, ctx, namespace.Config{Prefix: "isolation", Inject: true})
+			ns2 := namespace.NewOrFail(ctx, ctx, namespace.Config{Prefix: "isolation", Inject: true})
+
+			var a, b echo.Instance
+			echoboot.NewBuilderOrFail(ctx, ctx).
+				With(&a, echoConfig(ns1, "a")).
+				With(&b, echoConfig(ns2, "b")).
+				BuildOrFail(ctx)
+
+			time.Sleep(2 * time.Second)
+			baseline := cdsPushesOrFail(ctx)
+			scopes.Framework.Infof("cds pushes before sidecar: %d", baseline)
+
+			ctx.Config().ApplyYAMLOrFail(ctx, ns1.Name(), isolationSidecar)
+
+			time.Sleep(3 * time.Second)
+			after := cdsPushesOrFail(ctx)
+			scopes.Framework.Infof("cds pushes after sidecar: %d", after)
+
+			if after-baseline > 1 {
+				ctx.Fatalf("expected no more than 2 cds pushes when creating sidecar, got: %d", after-baseline)
+			}
+		})
+}
+
+func cdsPushesOrFail(ctx framework.TestContext) int {
```

Comment:
I suspect this will be flaky. any changes unrelated to the test (in another namespace, kube-system, etc) can impact this

Reply:
Do we expect we will get additional coverage here that TestProxyNeedsPush does not give us?

Reply:
I was trying to see if there was a way I could check the # of pushes for a specific sidecar or maybe check the "push reason".

Mostly I just wanted to make sure this worked on a real cluster and saw that there were consistently 2 pushes without the code change and 1 with the scoping.

Reply:
Generally we try to limit the stats that are labeled by connection because it grows super fast and can be stressful on prometheus (and istiod)